### PR TITLE
[TensorIF] Supports other/tensor(s) in output format

### PR DESCRIPTION
**Pull request by gichan-jang**
Supports other/tensor(s) in output format.

      - Supported operator: All
      - Supported data type: All
      - Supported action: passthrough, skip, tensorpick. (Others will be added)
      - Output format: other/tensor or other/tensors
    
Test pipeline description
```
gst-launch-1.0 v4l2src ! videoconvert ! videoscale ! video/x-raw, format=RGB, width=640, height=480, framerate=30/1 ! tensor_converter ! mux.sink_0 \
               videotestsrc ! videoconvert ! videoscale ! video/x-raw, format=RGB, width=640, height=480, framerate=30/1 ! tensor_converter ! mux.sink_1 \
               tensor_mux name=mux ! tensor_if name=tif \
                                               compared_value=A_VALUE compared-value-option=0:0:0:0,0 \
                                               supplied-value=70,150 \
                                               operator=RANGE_INCLUSIVE \
                                               then = PASSTHROUGH\
                                               else = TENSORPICK \
                                               else-option= 1 \
                tif.src_0 ! tensor_demux name=demux \
                    demux.src_0 ! tensor_decoder mode=direct_video ! videoconvert ! ximagesink async=false sync=false \
                    demux.src_1 ! tensor_decoder mode=direct_video ! videoconvert ! ximagesink async=false sync=false \
                tif.src_1 ! tensor_decoder mode=direct_video ! videoconvert ! ximagesink async=false sync=false
```

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

Self evaluation:
1. Build test: [ *]Passed [ ]Failed [ ]Skipped
2. Run test: [ *]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_if/gsttensorif.c [hunk 1]
```diff
@@ -31,12 +31,15 @@
...
- *      operator=RANGE_INCLUSIVE
- *      supplied_values=10,100
- *      then=PASSTHROUGH
- *      else=FILL_WITH_FILE else_option=${path_to_file}
- *    ! (tensor stream) ...
+ *      tensor_if name=tif \
+ *                compared-value=A_VALUE compared-value_option=3:4:2:5,0 \
+ *                operator=RANGE_INCLUSIVE \
+ *                supplied-values=10,100 \
+ *                then=PASSTHROUGH \
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_if/gsttensorif.c:38]: PASSTHROUGH == TENSOR_PICK at src.0? (need to document it, too)

File: gst/nnstreamer/tensor_if/gsttensorif.c [hunk 2]
```diff
@@ -31,12 +31,15 @@
...
+ *                else=TENSORPICK \
+ *                else-option=0 \
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_if/gsttensorif.c:40]: else-option=1 ?

File: gst/nnstreamer/tensor_if/gsttensorif.c [hunk 3]
```diff
@@ -49,15 +52,17 @@
...
- *        d.src_0 ! queue ! fakesink # throw away the 1/0 value.
- *        d.src_1 ! queue ! do whatever you want here...
+ *      ! tensor_if name=tif \
+ *          compared_value=A_VALUE \
+ *          compared_value_option=0:0:0:0,0 \# 1st tensor's [0][0][0][0].
+ *          operator=EQ \
+ *          supplied_values=1 \
+ *          then=PASSTHROUGH \# or whatsoever you want
+ *          else=SKIP \# or whatsoever you want
+ *      ! tif.src_0 ! tensor_demux name=d \ #for TRUE action
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_if/gsttensorif.c:62]: ```! tif.src_0``` is not required. (and looks dangerous)

File: gst/nnstreamer/tensor_if/gsttensorif.c [hunk 4]
```diff
@@ -31,12 +31,15 @@
...
- *    ! (tensor stream) ...
+ *      tensor_if name=tif \
+ *                compared-value=A_VALUE compared-value_option=3:4:2:5,0 \
+ *                operator=RANGE_INCLUSIVE \
+ *                supplied-values=10,100 \
+ *                then=PASSTHROUGH \
+ *                else=TENSORPICK \
+ *                else-option=0 \
+ *    ! tif.src_0 ! (tensor(s) stream for TRUE action) ...
+ *    ! tif.src_1 ! (tensor(s) stream for FALSE action) ...
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_if/gsttensorif.c:42]: ! is not required before labels.

File: gst/nnstreamer/tensor_if/gsttensorif.c [hunk 5]
```diff
@@ -49,15 +52,17 @@
...
+ *          compared_value=A_VALUE \
+ *          compared_value_option=0:0:0:0,0 \# 1st tensor's [0][0][0][0].
+ *          operator=EQ \
+ *          supplied_values=1 \
+ *          then=PASSTHROUGH \# or whatsoever you want
+ *          else=SKIP \# or whatsoever you want
+ *      ! tif.src_0 ! tensor_demux name=d \ #for TRUE action
+ *         d.src_0 ! queue ! fakesink # throw away the 1/0 value.
+ *         d.src_1 ! queue ! do whatever you want here...
+ *      ! tif.src_1 ! (tensor(s) stream for FALSE action) ...
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_if/gsttensorif.c:65]: If it is PASSTHROUGH and SKIP, how can we have src_1?

File: gst/nnstreamer/tensor_if/gsttensorif.c (hunk 2 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_if/gsttensorif.c:40]: It does not matter whether it is 0 or 1.

File: gst/nnstreamer/tensor_if/gsttensorif.c (hunk 4 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_if/gsttensorif.c:42]: Right. I will remove it

File: gst/nnstreamer/tensor_if/gsttensorif.c (hunk 5 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_if/gsttensorif.c:65]: Right. I will remove it and clear example script.

File: gst/nnstreamer/tensor_if/gsttensorif.c (hunk 3 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_if/gsttensorif.c:62]: Right. In this case it's not required.

File: gst/nnstreamer/tensor_if/gsttensorif.c (hunk 1 above)
Comment (gichan-jang) [reply gst/nnstreamer/tensor_if/gsttensorif.c:38]: src_0 is for TRUE action.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Review by myungjoo — CHANGES_REQUESTED** [review]
If output-stream's ```num_tensors == 1```, it can be both ```other/tensor``` and ```other/tensors```. Please support this requirement.

**Comment by gichan-jang** [comment]
> If output-stream's `num_tensors == 1`, it can be both `other/tensor` and `other/tensors`. Please support this requirement.

Updated to support both `other/tensor` and `other/tensors` when the number of tensors is 1.
If the peer element supports both tensor and tensors, the `other/tensor` has a higher priority

**Comment by myungjoo** [comment]
> > If output-stream's `num_tensors == 1`, it can be both `other/tensor` and `other/tensors`. Please support this requirement.
> 
> Updated to support both `other/tensor` and `other/tensors` when the number of tensors is 1.
> If the peer element supports both tensor and tensors, the `other/tensor` has a higher priority

Please add another PR that tests both cases (other/tensor and other/tensors of num_tensor=1)

**Comment by gichan-jang** [comment]
> > > If output-stream's `num_tensors == 1`, it can be both `other/tensor` and `other/tensors`. Please support this requirement.
> > 
> > 
> > Updated to support both `other/tensor` and `other/tensors` when the number of tensors is 1.
> > If the peer element supports both tensor and tensors, the `other/tensor` has a higher priority
> 
> Please add another PR that tests both cases (other/tensor and other/tensors of num_tensor=1)

Got it. I will add test cases for them.

**Review by myungjoo — APPROVED** [review]

**Pull request merged**